imagebuildah/stage_executor.go

```diff
@@ -627,6 +628,28 @@ func (s *StageExecutor) Run(run imagebuilder.Run, config docker.Config) error {
 		WorkingDir:       config.WorkingDir,
 	}
 
+	// Honor `RUN --network=<>`.
+	switch run.Network {
+	case "host":
+		didConfigure := false
+		for i, namespace := range options.NamespaceOptions {
+			if namespace.Name == "network" {
+				didConfigure = true
+				options.NamespaceOptions[i].Host = true
+			}
+		}
+		if !didConfigure {
+			options.NamespaceOptions = append(options.NamespaceOptions, define.NamespaceOption{Name: "network", Host: true})
```

Comment:
`options.NamespaceOptions` has an `AddOrReplace()` method which should simplify this a bit.

Reply:
Most of the logic in this `case` clause should be unnecessary now.

Reply:
Done, removed other logic